hasher-matcher-actioner/scripts/submit_content_test.py

```diff
@@ -0,0 +1,165 @@
+#! /usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import os
+import json
+import base64
+import requests
+import dataclasses
+import typing as t
+from urllib.parse import urljoin
+from script_utils import HasherMatcherActionerAPI
+
+from hmalib.common.evaluator_models import Action, ActionRule
+from hmalib.common.classification_models import Label, ActionLabel, ClassificationLabel
+from hmalib.common.actioner_models import ActionPerformer, WebhookGetActionPerformer
+
+
+class DeployedInstanceTestHelper:
+    """
+    Class around testing a deployed instance of HMA from Content Submission to Hash - Match - Action
+    by checking that the expected values are found
+    """
+
+    SAMPLE_PG_ID = "inria-holidays-test"
+
+    def __init__(
+        self,
+        api_url: str,
+        api_token: str = "",
+        client_id: str = None,
+        refresh_token: str = None,
+    ) -> None:
+        if not api_token and (not client_id or not refresh_token):
+            print(
+                "Test requires an api_token OR a client_id + refresh_token to function"
+            )
+            exit(1)
+
+        self.api = HasherMatcherActionerAPI(
+            api_url, api_token, client_id, refresh_token
+        )
+
+    def _refresh_token(self):
+        self.api._refresh_token()
+
+    def create_dataset_config(
+        self,
+        privacy_group_id: str,
+        privacy_group_name: str,
+        description: str = "",
+        matcher_active: bool = True,
+        fetcher_active: bool = False,
+        write_back: bool = False,
+    ):
+        self.api.create_dataset_config(
+            privacy_group_id,
+            privacy_group_name,
+            description,
+            matcher_active,
+            fetcher_active,
+            write_back,
+        )
+
+    def create_action(
+        self,
+        action_performer: ActionPerformer,
+    ):
+        self.api.create_action(
+            name=action_performer.name,
+            config_subtype=action_performer.get_config_subtype(),
+            fields={
+                key: value
+                for key, value in vars(action_performer).items()
+                if key not in {"name", "config_subtype"}
+            },
+        )
+
+    def create_action_rule(
+        self,
+        action_rule: ActionRule,
+    ):
+        # Need to give the api a json like dict object (just like is used in aws)
+        self.api.create_action_rule(action_rule.to_aws())
+
+
+def set_up(helper: DeployedInstanceTestHelper):
```

Comment:
Is there idempotency in this method? What if I call set_up twice?

Reply:
yes because the request fails if it already exists. I'll add a doc string